src/main/java/de/qabel/desktop/hockeyapp/VersionClient.java

```diff
@@ -0,0 +1,122 @@
+package de.qabel.desktop.hockeyapp;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.protocol.HTTP;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public class VersionClient {
+
+    public static final String API_VERSIONS_NEW = "/app_versions/new";
+    public static final String API_VERSIONS_ALL = "/app_versions";
+    private final HockeyAppConfiguration config;
+    public HttpClient httpClient;
+    private List<HockeyAppVersion> versions = new LinkedList<>();
+    private HockeyAppVersion version;
+
+    public VersionClient(HockeyAppConfiguration config, HttpClient httpClient) {
+        this.httpClient = httpClient;
+        this.config = config;
+    }
+
+
+    void loadVersions() throws IOException, JSONException {
+
+        HttpGet httpGet = config.getHttpGet(API_VERSIONS_ALL);
+
+        HttpResponse response = httpClient.execute(httpGet);
+        String responseContent = EntityUtils.toString(response.getEntity());
+
+        parseVersionsResponse(responseContent);
```

Comment:
You chain method calls a lot. That makes the program flow hard to follow because you have to jump into every method to see what it might modify. Try to work more with returns and functions (that don't modify the state).

**parseVersionsResponse** for example could to what it's named for. Parse the versions ... and then return them.
A `versions = parseVersionsResponse(responseContent)` is more obvious. You can see directly, that it modifies the versions. And it is btw. more thread safe because multiple threads would parse simultaniously and waste resources but you would get a consistent set of versions in `versions`. Now, if parseVersionsResponse is called multiple times at the same time, the results would get mixed and you would probably end up with duplicates.
loadVersions() could also return the versions... it would then "load" the versions from the web and return them ... straight purpose without an implicitly modified state. 

Oh and `loadVersions()` is never called but from a test ... whats up here?